CBot/src/CBot/context/list_constant_interface.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of the Colobot: Gold Edition source code
+ * Copyright (C) 2001-2023, Daniel Roux, EPSITEC SA & TerranovaTeam
+ * http://epsitec.ch; http://colobot.info; http://github.com/colobot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://gnu.org/licenses
+ */
+
+#pragma once
+
+#include "CBot/context/list_constant.h"
+
+namespace CBot
+{
+
+class CBotListConstantInterface
+{
+public:
+    virtual ~CBotListConstantInterface() {}
+
+    template<typename T>
+    bool AddConstant(const std::string& name, const T& value)
+    {
+        return m_listConstant.AddConstant(name, value);
+    }
+
+    virtual bool IsDefinedConstant(const std::string& name) const = 0;
+
+    virtual const CBotVarUPtr& GetDefinedConstant(const std::string& name) = 0;
```

Comment:
Why such an exotic return type? Just return a const reference to a CBotVar. How the ownership is managed is an implementation detail.

```suggestion
    virtual const CBotVar& GetDefinedConstant(const std::string& name) = 0;
```

The same applies to other classes where you define GetDefinedConstant()

Reply:
I was partially because if GetDefinedConstant() was misused and there is no constant defined with the ```name``` given, ```unordered_map::operator[]``` constructs an empty unique_ptr.

It also enables the shortened syntax: (which I didn't use in this case)
```c++
if ( const auto& var = GetDefinedConstant(name) )
{
    // do stuff using var because it's not an empty unique_ptr
}
```

Reply:
How about this?

```c++
    /**
     * \brief Find a constant by name
     *
     * Return nullptr if not found
     */
    virtual const CBotVar* GetDefinedConstant(const std::string& name) = 0;
```

```c++
if ( auto var = GetDefinedConstant(name) )
{
    // do stuff using *var
}
```

Reply:
About interpreting a raw pointer as a boolean expression to test if it's not nullptr;

I have been under the impression that this was mostly removed from the code, and that it's preferable to use ```x != nullptr``` explicitly.

Maybe the explicit syntax: ```if (auto* x = some_pointer) {...}``` can be an exeception to the rule.